swap/src/protocol/alice/behaviour.rs

```diff
@@ -53,8 +53,10 @@ impl From<alice::OutEvent> for OutEvent {
 
 impl From<execution_setup::OutEvent> for OutEvent {
     fn from(event: execution_setup::OutEvent) -> Self {
+        use crate::protocol::alice::execution_setup::OutEvent::*;
         match event {
-            execution_setup::OutEvent::Done(res) => OutEvent::ExecutionSetupDone(res.map(Box::new)),
+            Done(state3) => OutEvent::ExecutionSetupDone(Box::new(state3)),
+            Failure(err) => OutEvent::Failure(err),
```

Comment:
Thank for pointing this out.

Originally, this change was needed because `anyhow::Error` is not `Clone` and hence cannot be in a `broadcast::Channel`. However, I removed the channel that carries `anyhow::Result<State3>` so I could have removed this refactoring.

I kept the refactoring because I just wanted to align the result processing with the other behaviours by gathering all failures in the `OutEvent` of the "root" `Behaviour`. IMHO, having this odd case of the event returning a `Result` without a proper justification for it would be confusing for future readers.